libr/core/carg.c

```diff
@@ -0,0 +1,248 @@
+/* radare - LGPL - Copyright 2009-2018 - pancake */
+
+#include <r_core.h>
+
+#define MAXSTRLEN 50
+
+static void set_fcn_args_info(RAnalFuncArg *arg, RAnal *anal, const char *fcn_name, const char *cc, int arg_num) {
+	if (!fcn_name || !arg || !anal) {
+		return;
+	}
+	arg->name = r_anal_type_func_args_name (anal, fcn_name, arg_num);
+	arg->orig_c_type = r_anal_type_func_args_type (anal, fcn_name, arg_num);
+	if (!strncmp ("const ", arg->orig_c_type, 6)) {
+		arg->c_type = arg->orig_c_type + 6;
+	} else {
+		arg->c_type = arg->orig_c_type;
+	}
+	const char *query = sdb_fmt (-1, "type.%s", arg->c_type);
```

Comment:
Keep in mind there are can be a complex types like structs and unions. Even if you do not display them - you need to handle those cases correctly still.

Reply:
Yup lot of work to do , also function argument analysis needs more improvement ,  i have a few in my mind , i could do that whenever time permits , that's why for now i have disabled output by default and u have to use the config var 'dgb.funcarg' to enable it